### PR TITLE
Allow passing relative paths instead of absolute paths

### LINKED ISSUE #7: Prepend CWD to other options instead of making developers pass absolute path to every option.

### PR DESCRIPTION
**Pull request by smartclash**
PRs #3 and #4 require developers to pass absolute path to typings file or test files even though they have already provided the current working directory. This PR fixes #7 and allows us to pass relative paths.

Be sure to just go through #3 and #4 as they have the main changes. This PR is more like a fine-tuned final version of those.

By merging this, TSD can be used in many ways

**Method 1:**

Let TSD figure out everything by itself. Uses `process.cwd()`
```js
const diagnose = await tsd();
```
**Method 2:**

Pass `cwd` and also the relative path to `typingsFile` only.
```js
const diagnose = await tsd({
  cwd: path.join(__dirname, '/custom/path'),
  typingsFile: 'someName.d.ts'
});
```
**Method 3:**

Pass `cwd` and also the relative path to `testFiles`.
```js
const diagnose = await tsd({
  cwd: path.join(__dirname, '/custom/path'),
  testFiles: [
    '/somewhere/app.test.js',
    '/somewhere/else/math.test.js'
  ]
});
```

**Method 4:**

Pass `cwd` and pass relative paths to both `typingsFile` and `testFiles`.
```js
const diagnose = await tsd({
  cwd: path.join(__dirname, '/custom/path'),
  typingsFile: 'someName.d.ts',
  testFiles: [
    '/somewhere/app.test.js',
    '/somewhere/else/math.test.js'
  ]
});
```

### CODE REVIEW

File: source/lib/index.ts [hunk 1]
```diff
@@ -68,11 +72,11 @@ export default async (options: Options = {cwd: process.cwd()}) => {
 	});
 
 	const context: Context = {
-		cwd: options.cwd,
```
Comment (jevakallio) [inline source/lib/index.ts]: Seems like this is just reordering props? Has anything actually changed here?

File: tsconfig.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
 		"noUnusedParameters": true,
 		"noFallthroughCasesInSwitch": true,
 		"noEmitOnError": true,
+		"sourceMap": true,
```
Comment (jevakallio) [inline tsconfig.json]: Is this an intentional change? Seems unrelated to the PR overall

File: source/lib/index.ts [hunk 3]
```diff
@@ -56,10 +60,10 @@ export default async (options: Options = {cwd: process.cwd()}) => {
 	}
 
 	const pkg = pkgResult.packageJson;
-
 	const config = loadConfig(pkg as any, options.cwd);
 
-	// Look for a typings file, otherwise use `index.d.ts` in the root directory. If the file is not found, throw an error.
+	// Look for a typings file if not explicitly specified, otherwise use `index.d.ts` in the root directory.
```
Comment (jevakallio) [inline source/lib/index.ts]: The comment has changed but the implementation hasn't. Was the old comment incorrect?

**Review by SaurabhAgarwala — APPROVED** [review]
LGTM
